pkg/apis/hive/v1/clusterdeployment_types.go

```diff
@@ -181,7 +170,7 @@ type ClusterDeploymentStatus struct {
 
 	// FederatedClusterRef is the reference to the federated cluster resource associated with
 	// this ClusterDeployment.
-	FederatedClusterRef *corev1.ObjectReference `json:"federatedClusterRef,omitempty"`
+	FederatedCluster *corev1.ObjectReference `json:"federatedCluster,omitempty"`
```

Comment:
Ugh we should remove this entirely, even if we keep the code it should be axed from the V1 API until it's useful.

Reply:
I'll remove the federation stuff as its own PR.